### ISSUE TITLE
[Feature request] create subsets of fonts

### ISSUE THREAD

**Issue by perrin4869** [agentk/fontfacegen#9]
In languages like Japanese fonts can be several megs big. It would be really useful to have a feature akin to fontsquirrel subset here. Would be able to automatically generate fonts with the characters required and only those characters. For your consideration :)

**Comment by agentk** [comment]
That is a really good idea. I'm not sure where to start to work out how to do it though. I'm open to suggestions.

**Comment by perrin4869** [comment]
Well, I contacted fontsquirrel and they said that in their backend they are using fontforge. Then I came across this article, in japanese: http://itouhiro.hatenablog.com/entry/20140511/font
Basically what they are doing is, they write a script that gets a list of used characters inside a certain file (`listChar.js`), and then they use another script, `listCharFF.js`, where `FF` probably stands for `FontForge`, to create the following file as an example: 

```SelectWorthOutputting();
SelectFewer(0ud7);
SelectFewer(0u2015);
SelectFewer(0u2026);
SelectFewer(0u25cb);
SelectFewer(0u3000);
SelectFewer(0u3001);
SelectFewer(0u3002);
SelectFewer(0u3005);
SelectFewer(0u300a);
...
```

They run that inside fontforge and that creates a font subset.

**Comment by dandaka** [comment]
+1 for this feature. A must have, because loading time with webfonts increases dramatically.

We could look into how this feature is working with existing solutions:

1. Font Squirrel
http://www.fontsquirrel.com/tools/webfont-generator

![font-squirrel](https://cloud.githubusercontent.com/assets/2384694/12075664/53cdd1c6-b18f-11e5-9add-48e1a4ddcf43.jpg)

It gets down to Unicode ranges. But for a good UX, it is better to have meaningful options for each case.

BTW, Font Squirrel options aren't correct all the times (or maybe I don't understand a logic behind). For example, `[X] Cyrillic letters` gives me Latin letters also. Character type is also wrong, `[X] Lowercase` gives lowercase latin.

2. Fontie
https://fontie.flowyapps.com/home

Here is Unicode ranges from their options
https://gist.github.com/dandaka/e788bf811f8f2270dafc

This is less useful for me. I have killed lots of time, trying to find Polish letters "ść" there. Font Squirrel options are definitely better, but I can't find ranges in public. Will try to contact developers of FS.

**Comment by dandaka** [comment]
I think we could start with simple "custom Unicode ranges" option. And later find a way to transform meaningful Font Squirrel subset settings to Unicode ranges.

**Comment by perrin4869** [comment]
I came across this recently:

https://github.com/shane-tomlinson/connect-fonts-tools
https://github.com/shane-tomlinson/connect-fonts-tools/blob/master/scripts/subset

Basically it's a wrapper around fontforge for font subset creating in node.js. Unfortunately there is no API, it's just a command-line tool, but the eventual implementation of subsetting here could be based off of that code for simplicity sake

**Comment by agentk** [comment]
Great find. I would love to be able to get support for subsetting built in. I will talk to Shane and work out the best way.

**Comment by perrin4869** [comment]
I am thinking of attempting to put together a PR for this this weekend. Any thoughts?

**Comment by sgen** [comment]
Give it a shot and well see what the resulting PR looks like. This appears to be a very complex feature, so there my be some kinks to iron out.

**Comment by perrin4869** [comment]
Released v0.2.0
